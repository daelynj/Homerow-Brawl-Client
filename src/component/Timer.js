import React, { useState, useEffect } from 'react';

function Timer(props) {
  const [seconds, setSeconds] = useState(0);

  useEffect(() => {
    var timerID = setTimeout(tick, 1000 );

    return function cleanup () {
      props.finishTime(seconds);
    }
  });

  function tick() {
    setSeconds(seconds + 1);
  }

  return (
    <div>
      Timer: {seconds}
    </div>
  );
}

export default Timer;
