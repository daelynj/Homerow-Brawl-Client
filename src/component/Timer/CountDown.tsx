import * as React from "react";
import { CountDownButton } from "./button/CountDownButton";
import { CountDownTimer } from "./CountDownTimer";

interface Props {
  setCountDown: (newCountDown: any) => void;
  setCountUp: (newCountUp: any) => void;
  updateCountDown: (updateCountDown: any) => void;
  countDown: boolean;
  countUp: boolean;
}

export const CountDown = (props: Props) => {
  const renderCountDownTimer = () => (
    <CountDownTimer setCountUp={props.setCountUp} />
  );

  const renderButton = () => (
    <CountDownButton
      updateCountDown={props.updateCountDown}
      setCountDown={props.setCountDown}
    />
  );

  const shouldRenderButton = () => (props.countDown ? false : true);

  return (
    <div>
      {shouldRenderButton() && renderButton()}
      {!props.countUp && !shouldRenderButton() && renderCountDownTimer()}
    </div>
  );
};
