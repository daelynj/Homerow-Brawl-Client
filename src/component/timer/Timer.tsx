import * as React from "react";
import { CountDownButton } from "./button/CountDownButton";
import { CountDownTimer } from "./CountDownTimer";
import { CountUpTimer } from "./CountUpTimer";

interface Props {
  setCountDown: (newCountDown: boolean) => void;
  setCountUp: (newCountUp: boolean) => void;
  updateCountDown: (updateCountDown: boolean) => void;
  countDown: boolean;
  countUp: boolean;
  setFinishTime: (newFinishTime: number) => void;
}

export const Timer = (props: Props) => {
  const renderCountDownTimer = () => (
    <CountDownTimer setCountUp={props.setCountUp} />
  );

  const renderCountUpTimer = () => (
    <CountUpTimer setFinishTime={props.setFinishTime} />
  );

  const handleEvent = () => {
    props.setCountDown(true);
    props.updateCountDown(true);
  };

  return (
    <>
      {!props.countDown && <CountDownButton handleEvent={handleEvent} />}
      {!props.countUp && props.countDown && renderCountDownTimer()}
      {props.countUp && renderCountUpTimer()}
    </>
  );
};
