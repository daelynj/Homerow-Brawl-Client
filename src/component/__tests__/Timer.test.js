import React from 'react';
import Timer from '../Timer';
import renderer from 'react-test-renderer';

describe('Timer', () => {
  jest.useFakeTimers();
  const finishTime = jest.fn();

  it('renders the Timer', () => {
    const tree = renderer.create(<Timer/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls tick every 1000 miliseconds', () => {
    const TimerComponent = renderer.create(<Timer time = {0} finishTime = {finishTime}/>);

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);
    expect(setTimeout).toHaveBeenCalledTimes(1);

    TimerComponent.update();

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);
    expect(setTimeout).toHaveBeenCalledTimes(2);
  });

  it('sets the finish time when unmounted', () => {
    const TimerComponent = renderer.create(<Timer time = {0} finishTime = {finishTime}/>);
    TimerComponent.unmount();

    expect(finishTime).toHaveBeenCalled();
  });
});
