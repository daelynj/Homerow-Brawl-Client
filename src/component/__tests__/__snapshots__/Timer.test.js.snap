// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer renders the Timer 1`] = `
<div>
  Timer: 
</div>
`;
